services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/TransferManagerFactory.java

```diff
@@ -52,11 +52,8 @@ public static S3TransferManager createTransferManager(DefaultBuilder tmBuilder)
             return new CrtS3TransferManager(transferConfiguration, s3AsyncClient, isDefaultS3AsyncClient);
         }
 
-        if (s3AsyncClient.getClass().getName().equals("software.amazon.awssdk.services.s3.DefaultS3AsyncClient")) {
-            log.warn(() -> "The provided DefaultS3AsyncClient is not an instance of S3CrtAsyncClient, and thus multipart"
-                           + " upload/download feature is not enabled and resumable file upload is not supported. To benefit "
-                           + "from maximum throughput, consider using S3AsyncClient.crtBuilder().build() instead.");
-        } else {
+        if (!s3AsyncClient.getClass().getName()
+                          .equals("software.amazon.awssdk.services.s3.internal.multipart.MultipartS3AsyncClient")) {
```

Comment:
How about also updation the log message to mention with something like:
> The provided S3AsyncClient is neither an instance of S3CrtAsyncClient or MultipartS3AsyncClient, and thus multipart upload/download feature may not be enabled and resumable file upload may not be supported.
To benefit from maximum throughput, consider either using S3AsyncClient.crtBuilder().build() or S3AsyncClient.builder().multipartEnabled(true).build() instead.

Reply:
Changed.